extensions/extensionservice/org.eclipse.smarthome.extensionservice.marketplace/src/main/java/org/eclipse/smarthome/extensionservice/marketplace/MarketplaceExtensionHandler.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.extensionservice.marketplace;
+
+/**
+ * This interface can be implemented by services that want to register as handlers for specific marketplace extension
+ * types and formats.
+ * In a system there should always only be exactly one handler responsible for a given type+format combination. If
+ * multiple handers support it, it is undefined which one will be called.
+ * This mechanism allows solutions to add support for specific formats (e.g. Karaf features) that are not supported by
+ * ESH out of the box.
+ * It also allows to decide which extension types are made available at all.
```

Comment:
The types are statically defined in the IoT marketplace - so if the marketplace is extended, we should also maintain the new types here at a central place as constants.